libparsec/crates/client/src/workspace/merge.rs

```diff
@@ -308,30 +297,28 @@ pub(super) fn merge_local_folder_manifest(
     // is called `merge_in_progress`: if the local manifest contains other changes, then
     // we must do a proper children merge and update the manifest accordingly.
 
-    let mut merge_in_progress =
-        LocalFolderManifest::from_remote_with_restored_local_confinement_points(
-            remote,
-            prevent_sync_pattern,
-            local,
-            timestamp,
-        );
-    let remote = &merge_in_progress.base;
+    let mut unconfined_local = UnconfinedLocalFolderManifest::remove_confinement(local);
+    let mut unconfined_remote = UnconfinedLocalFolderManifest::from_remote(remote);
 
-    // 4) Shortcut in case only the remote has changed
-    if !local.need_sync {
-        // Note a weird corner case here:
-        // If the prevent sync pattern has changed but the local manifest hasn't had
-        // time to be updated yet, then it's possible to end up with
-        // `local_need_sync == false`, but `merge_in_progress.need_sync == true` !
-
-        // TODO: determine if that's an issue !!!
-        //       (also see https://github.com/Scille/parsec-cloud/issues/7885)
-        return MergeLocalFolderManifestOutcome::Merged(Arc::new(merge_in_progress));
+    // 3) Shortcut in case only the remote has changed
+    if !unconfined_local.need_sync {
+        let new_manifest =
+            unconfined_remote.apply_confinement(local, prevent_sync_pattern, timestamp);
+        return MergeLocalFolderManifestOutcome::Merged(Arc::new(new_manifest));
     }
 
     // Both the remote and the local have changed
 
-    // 5) The remote changes are ours (our current local changes occurs while
+    // The following case was present in the original v2 code.
+    // TODO: investigate why it's not present anymore and check whether
+    // it is still relevant.
+    // ```python
+    // # All the local changes have been successfully uploaded
+    // if local_manifest.match_remote(remote_manifest):
+    //     return local_from_remote
+    // ```
```

Comment:
Is it something you will investigate during this PR or at a later time ?

If it's the later, could you create an issue and add its link in the TODO command ?